search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+        "_embedded" : {
+          "dspaceObject" : {
+            "uuid": "ff7ec3a4-0aab-418b-94fc-d0e8189084db",
+            "name": "Test Item with no hit highlights",
+            "handle": "10673/5"
+          }
+        }
+      }
+    ]
+  },
+  "_links": {
+      "first": {
+        "href": "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&page=0&size=5"
+      },
+      "self": {
+        "href": "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&page=0&size=5"
+      },
+      "next": {
+        "href": "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&page=1&size=5"
+      },
+      "last": {
+        "href": "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&page=2&size=5"
+      }
+  }
+}
+```
+
+### Matching facet search results
+**/api/discover/search/<:dso-type>/facets**
+
+This endpoint returns a list of facet values and counts based on the current query. The result can be refined using the following parameters:
+* `query`: The discovery search string to will be used to match records.
+* `scope`: UUID of a specific DSpace container (site, community or collection) to which the search has to be limited, e.g. `scope=9076bd16-e69a-48d6-9e41-0238cb40d863`.
+* `<:filter-name>.<:filter-operator>`: Advanced search filter that has to be used to filter the result set. The `filter-name` and `filter-operator` must match a value returned by parent search endpoint (see above). For example `author.authority=5df05073-3be7-410d-8166-e254369e4166` or `title.notcontains=rainbows`.
+* There is no paging here since the number of facet values displayed in the Discovery search screen is configured in the backend.
+
+Example: TODO
+
+The returned JSON response will be like:
+
+```json
+{
+  "query":"my query",
+  "scope":"9076bd16-e69a-48d6-9e41-0238cb40d863",
+  "filters": [
+      {
+        "filter" : "title",
+        "operator" : "notcontains",
+        "value" : "abcd",
+        "label" : "abcd"
+      },
+      {
+        "filter" : "author",
+        "operator" : "authority",
+        "value" : "1234",
+        "label" : "Smith, Donald"
+      }
+  ],
+  "sort" :
+  {
+    "by" : "dc.title",
+    "order" : "asc"
+  },
+  "_embedded" : {
+    "facets" : [
+      {
+        "name" : "author",
+        "_embedded" : {
+          "values" : [
+              {
+                "value" : "Smith, Donald 2",
+                "count" : 100,
+                "_links": {
+                  "search" : "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&author.equals=Smith,+Donald+2"
```

Comment:
This is the same link as used in the `/api/discover/facets/<:facet-name>` endpoint where it is actually a search for the items related to that facet value.